numpy/_core/src/multiarray/descriptor.c

```diff
@@ -726,32 +726,34 @@ _convert_from_list(PyObject *obj, int align)
 static PyArray_Descr *
 _convert_from_commastring(PyObject *obj, int align)
 {
-    PyObject *listobj;
+    PyObject *parsed;
     PyArray_Descr *res;
-    PyObject *_numpy_internal;
+    static PyObject *_numpy_internal = NULL;
     assert(PyUnicode_Check(obj));
-    _numpy_internal = PyImport_ImportModule("numpy._core._internal");
     if (_numpy_internal == NULL) {
-        return NULL;
+        _numpy_internal = PyImport_ImportModule("numpy._core._internal");
+        if (_numpy_internal == NULL) {
+            return NULL;
+        }
     }
-    listobj = PyObject_CallMethod(_numpy_internal, "_commastring", "O", obj);
-    Py_DECREF(_numpy_internal);
-    if (listobj == NULL) {
+    parsed = PyObject_CallMethod(_numpy_internal, "_commastring", "O", obj);
+    if (parsed == NULL) {
         return NULL;
     }
-    if (!PyList_Check(listobj) || PyList_GET_SIZE(listobj) < 1) {
-        PyErr_SetString(PyExc_RuntimeError,
-                "_commastring is not returning a list with len >= 1");
-        Py_DECREF(listobj);
-        return NULL;
+    if (PyUnicode_Check(parsed)
+        || (PyTuple_Check(parsed) && PyTuple_GET_SIZE(parsed) == 2)) {
+        res = _convert_from_any(parsed, align);
     }
-    if (PyList_GET_SIZE(listobj) == 1) {
-        res = _convert_from_any(PyList_GET_ITEM(listobj, 0), align);
+    else if (PyList_Check(parsed) && PyList_GET_SIZE(parsed) >= 1) {
```

Comment:
```suggestion
    else if (PyList_Check(parsed)) {
```
I suppose length 0 is impossible here anyway?  I suspect we could remove the size check.

Reply:
I think the length checks were mainly meant to guard against silly mistakes in the python code. My sense would be to leave it in.